mantidproject/mslice
# Refactor plotting to use mantid projection

- All plotting is now done directly from workspace objects.
- Caching of slice-specific workspace data e.g. intensities is saved in `SliceCache` class.
- Plot window is deleted when closed, so state of checkboxes etc is not saved.

**To test:**
To test MSlice PRs copy the latest commit SHA1 hash (here it is `f75fe63d567a0c10bfd4609390c05fb2487ae3f4` at the moment) and paste it into [scripts/ExternalInterfaces/CMakeLists.txt](https://github.com/mantidproject/mantid/blob/1bb2c9824772b1d1fb1efaea190c03389f7b5700/scripts/ExternalInterfaces/CMakeLists.txt#L9). CMake should pull this commit when you rebuild mantid.

- Load a workspace  [MAR21335_Ei60.00meV.nxs.zip](https://github.com/mantidproject/mslice/files/2269886/MAR21335_Ei60.00meV.nxs.zip)




Test Slice
- Under 'Slice', enter some values in the intensity fields
- Display the slice. The entered values should be the limits of the colorbar.
- Check all of the options in the Information and Intensity menus at the top work without errors.
- Check that an overplotted line's appearance can be changed by double-clicking it.
- Close and reopen slice window, then check that any options selected in the Information/Intensity menus have been cleared.
- Check that clicking interactive cuts and drawing a rectangle produces a cut window correctly.

Test Cut
- Click the cut tab, with a workspace selected enter some values for the first and second axis (not width).
- Check that a line appearance can be changed by double-clicking
- With the plot still open, modify the values in the cut tab and click 'plot over.' Line should appear on the same plot and legend should be updated.
- Check that the axis ranges and labels can be edited by double-clicking on them.





Fixes #358
